compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -547,8 +547,22 @@ object TreeChecker {
         i"owner chain = ${tree.symbol.ownersIterator.toList}%, %, ctxOwners = ${ctx.outersIterator.map(_.owner).toList}%, %")
     }
 
+    private def checkParents(tree: untpd.TypeDef)(using Context): Unit = {
+      val TypeDef(_, impl: Template) = tree: @unchecked
+      assert(ctx.owner.isInstanceOf[ClassSymbol])
+      val symbolParents = ctx.owner.asClass.info.parents
+      val treeParents = impl.parents.map(_.tpe)
+      assert(symbolParents.toSet == treeParents.toSet,
```

Comment:
```suggestion
      assert(symbolParents== treeParents,
```

The order of the parents does matter for member resolution.